[Feature] Detect Headphone + Label

ahammer opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* To show "Headset" when a wired headset is plugged (Instead of "Android")
* Update a deprecated API
* Fix bug with Android 12 runtime permission and a pre-connected device on first-launch

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[X] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

iaulakh commented:
Adam, noticed few bugs:
1) on fresh install if bluetooth is connected before, in audio device selection device is not displayed

<img width="475" alt="Screen Shot 2022-03-14 at 7 49 19 AM" src="https://user-images.githubusercontent.com/73618019/158093866-08ffff73-8701-4cde-aa7e-a8a09a9efff9.png">

2) Audio device list displays device name but button says bluetooth
<img width="449" alt="Screen Shot 2022-03-14 at 7 48 04 AM" src="https://user-images.githubusercontent.com/73618019/158093854-53aa8776-9a1b-42b9-bc6c-b0b13321cd5d.png">

ahammer commented:
@iaulakh 
- Added check for permission approval to init bluetooth (for android 12 Runtime permissions for bluetooth).
- Fixed SetupControlBarView (Shows device name & Headset when plugged)
- Removed Comment from last PR (from merge commit)
- getAdapter() instead of getDefaultAdapter() (deprecated update)

iaulakh commented on azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/common/audiodevicelist/AudioDeviceListViewModel.kt:
```diff
@@ -14,6 +15,11 @@ internal class AudioDeviceListViewModel(
     private val dispatch: (Action) -> Unit,
 ) {
 
+    val receiverTypeTitle get() = when (audioStateFlow.value.isHeadphonePlugged) {
```
Adam, it will be great if we can avoid view classes in view model.

ahammer replied:
Is string resource id considered a view class? I can push down into the view.

iaulakh reviewed (changes requested):
Adam, I noticed a leftover memory. We can try to find the cause. Tested, looks good
<img width="1397" alt="Screen Shot 2022-03-14 at 12 34 50 PM" src="https://user-images.githubusercontent.com/73618019/158121500-c3763964-8c2e-40c9-8684-00e3245ff4c1.png">

ahammer commented:
@iaulakh I think good now (no memory leak) small fix for that.

phileo99 reviewed (dismissed):
tested BT, ran all UI tests
👍
